test/utility/GridTools.h

```diff
@@ -48,82 +46,56 @@ struct gridtools_storage_types {
-                                             gridtools::aligned<cpu_alignment>>;
-  using gpu_2d_meta_data_type =
-      storage_traits_type::meta_storage_type<4, gpu_2d_layout_type, halo_3d_type,
-                                             gridtools::aligned<gpu_alignment>>;
-
-  using cpu_3d_meta_data_type =
-      storage_traits_type::meta_storage_type<5, cpu_3d_layout_type, halo_3d_type,
-                                             gridtools::aligned<cpu_alignment>>;
-  using gpu_3d_meta_data_type =
-      storage_traits_type::meta_storage_type<6, gpu_3d_layout_type, halo_3d_type,
-                                             gridtools::aligned<gpu_alignment>>;
-
-  using cpu_4d_meta_data_type =
-      storage_traits_type::meta_storage_type<7, cpu_4d_layout_type, halo_4d_type,
-                                             gridtools::aligned<cpu_alignment>>;
-  using gpu_4d_meta_data_type =
-      storage_traits_type::meta_storage_type<8, gpu_4d_layout_type, halo_4d_type,
-                                             gridtools::aligned<gpu_alignment>>;
-
-  //===----------------------------------------------------------------------------------------===//
-  //     Storage
-  //===----------------------------------------------------------------------------------------===//
-  using cpu_2d_real_storage_type = storage_traits_type::storage_type<T, cpu_2d_real_meta_data_type>;
-  using gpu_2d_real_storage_type = storage_traits_type::storage_type<T, gpu_2d_real_meta_data_type>;
-  using cpu_2d_storage_type = storage_traits_type::storage_type<T, cpu_2d_meta_data_type>;
-  using gpu_2d_storage_type = storage_traits_type::storage_type<T, gpu_2d_meta_data_type>;
-
-  using cpu_3d_storage_type = storage_traits_type::storage_type<T, cpu_3d_meta_data_type>;
-  using gpu_3d_storage_type = storage_traits_type::storage_type<T, gpu_3d_meta_data_type>;
-
-  using cpu_4d_storage_type = storage_traits_type::storage_type<T, cpu_4d_meta_data_type>;
-  using gpu_4d_storage_type = storage_traits_type::storage_type<T, gpu_4d_meta_data_type>;
+      storage_traits_type::special_storage_info_t<3, gridtools::selector<1, 1, 0>, halo_3d_type>;
+  using gpu_2d_meta_data_type = gridtools::host_storage_info<4, gpu_2d_layout_type, halo_3d_type,
+                                                             gridtools::alignment<gpu_alignment>>;
+  using cpu_3d_meta_data_type = storage_traits_type::storage_info_t<5, 3, halo_3d_type>;
+  using gpu_3d_meta_data_type = gridtools::host_storage_info<6, gpu_3d_layout_type, halo_3d_type,
+                                                             gridtools::alignment<gpu_alignment>>;
+  using cpu_4d_meta_data_type = storage_traits_type::storage_info_t<7, 4, halo_4d_type>;
+  using gpu_4d_meta_data_type = gridtools::host_storage_info<8, gpu_4d_layout_type, halo_4d_type,
+                                                             gridtools::alignment<gpu_alignment>>;

```

Comment:
why is `host_storage_info` used and the using decl is named `gpu_4d_meta_data_type`?